### PR TITLE
[MNG-8179] Upgrade Parent to 43

### PR DESCRIPTION
**Pull request by slawekjaranowski**
https://issues.apache.org/jira/browse/MNG-8179

### CODE REVIEW

File: maven-api-impl/src/main/java/org/apache/maven/internal/impl/DefaultToolchainsXmlFactory.java [hunk 1]
```diff
@@ -53,10 +52,6 @@ public PersistedToolchains read(@Nonnull XmlReaderRequest request) throws XmlRea
             throw new IllegalArgumentException("reader or inputStream must be non null");
         }
         try {
-            InputSource source = null;
-            if (request.getModelId() != null || request.getLocation() != null) {
-                source = new InputSource(request.getModelId(), request.getLocation());
-            }
```
Comment (michael-o) [inline maven-api-impl/src/main/java/org/apache/maven/internal/impl/DefaultToolchainsXmlFactory.java:59]: This is weird, source is never used? Was this found part of a new rule/detection?

File: maven-core/src/test/resources/project-dynamism/pom.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+<project
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 
   <modelVersion>4.0.0</modelVersion>
```
Comment (michael-o) [inline maven-core/src/test/resources/project-dynamism/pom.xml:6]: Is this change intended? all of these POMs were invalid because they aren't in the target namespace and if you don't sue the `xsi` NS why declare it?

File: maven-embedder/src/main/java/org/fusesource/jansi/Ansi.java [hunk 3]
```diff
@@ -25,10 +25,10 @@
  * <a href="https://en.wikipedia.org/wiki/ANSI_escape_code#CSI_sequences">ANSI escape sequences</a>.
  *
  * This class comes from Jansi and is provided for backward compatibility
- * with maven-shared-utils, while Maven has migrated to JLine (into which Jansi has been merged
+ * wi§th maven-shared-utils, while Maven has migrated to JLine (into which Jansi has been merged
```
Comment (michael-o) [inline maven-embedder/src/main/java/org/fusesource/jansi/Ansi.java:28]: Typo

File: pom.xml [hunk 4]
```diff
@@ -24,7 +24,7 @@ under the License.
     <groupId>org.apache.maven</groupId>
     <artifactId>maven-parent</artifactId>
     <!-- when upgrading parent, also upgrade maven-bom's parent -->
-    <version>42</version>
+    <version>43</version>
```
Comment (michael-o) [inline pom.xml:27]: Nothing else? No property removal/override?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-api-impl/src/main/java/org/apache/maven/internal/impl/DefaultToolchainsXmlFactory.java (hunk 1 above)
Comment (slawekjaranowski) [reply maven-api-impl/src/main/java/org/apache/maven/internal/impl/DefaultToolchainsXmlFactory.java:59]: Yes new rule was detected it.
@gnodet - why is not used here ....?

Comment (gnodet) [reply maven-api-impl/src/main/java/org/apache/maven/internal/impl/DefaultToolchainsXmlFactory.java:59]: Toolchains do not currently fully support location tracking. I'll see if that can be done.

**Review by gnodet — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
@michael-o any more comments?

File: maven-core/src/test/resources/project-dynamism/pom-relative.xml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+<project
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
```
Comment (michael-o) [inline maven-core/src/test/resources/project-dynamism/pom-relative.xml:3]: My comments regarding this broken POMs still stand. No namespace means an invalid POM logically.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-core/src/test/resources/project-dynamism/pom-relative.xml (hunk 5 above)
Comment (gnodet) [reply maven-core/src/test/resources/project-dynamism/pom-relative.xml:3]: Is there any reason why they don't use the correct namespace ?

Comment (slawekjaranowski) [reply maven-core/src/test/resources/project-dynamism/pom-relative.xml:3]: ehh.. my IDE try to be more smart then I

Comment (slawekjaranowski) [reply maven-core/src/test/resources/project-dynamism/pom-relative.xml]: now - should be ok

**Review by michael-o — APPROVED** [review]
